test/vfd_swmr.c

```diff
@@ -3424,6 +3686,701 @@ test_enable_disable_eot(void)
+        FAIL_STACK_ERROR;
+
+    /* Close the checksum file */
+    if (chk_fp && HDfclose(chk_fp) != 0)
+        FAIL_STACK_ERROR;
+
+    if (buf)
+        HDfree(buf);
+
+    return 0;
+
+error:
+    if (buf)
+        HDfree(buf);
+    if (md_fp)
+        HDfclose(md_fp);
+    if (chk_fp)
+        HDfclose(chk_fp);
+
+    return -1;
+} /* md_ck_cb() */
+
+/*-------------------------------------------------------------------------
+ * Function:    test_updater_generate_md_checksums()
+ *
+ * Purpose:     It enables the generation of checksums for the metadata file
+ *              created by the writer end of tick function.
+ *              It also verifies the contents of the checksum file and the
+ *              updater files.
+ *
+ *              The test is invoked when the file is created via H5Fcreate()
+ *              and via H5Fopen().
+ *
+ * Return:      0 if test is sucessful
+ *              1 if test fails
+ *
+ * Programmer:  Vailin Choi; October 2021
+ *
+ * Note: It is important to clean up the checksum file and the updater files.
+ *
+ *-------------------------------------------------------------------------
+ */
+static unsigned
+test_updater_generate_md_checksums(hbool_t file_create)
+{
+    hid_t                   fid  = -1; /* File ID */
+    hid_t                   fcpl = -1; /* File creation property list ID */
+    hid_t                   fapl = -1; /* File access property list ID */
+    H5F_vfd_swmr_config_t   config;    /* Configuration for VFD SWMR */
+    H5F_generate_md_ck_cb_t cb_info;   /* Callback */
+
+    if (file_create) {
+        TESTING("VFD SWMR updater generate checksums for metadata file with H5Fcreate");
+    }
+    else {
+        TESTING("VFD SWMR updater generate checksums for metadata file with H5Fopen");
+    }
+
+    /* config, tick_len, max_lag, writer, maintain_metadata_file, generate_updater_files,
+     * flush_raw_data, md_pages_reserved, md_file_path, updater_file_path */
+    init_vfd_swmr_config(&config, 4, 7, TRUE, TRUE, TRUE, TRUE, 2, MD_FILE, UD_FILE);
+
+    /* use_latest_format, use_vfd_swmr, only_meta_page, page_buf_size, config */
+    fapl = vfd_swmr_create_fapl(TRUE, TRUE, FALSE, 4096, &config);
+
+    if (fapl == H5I_INVALID_HID)
+        TEST_ERROR
+
+    if ((fcpl = vfd_swmr_create_fcpl(H5F_FSPACE_STRATEGY_PAGE, 4096)) < 0) {
+        HDprintf("vfd_swmr_create_fcpl() failed");
+        FAIL_STACK_ERROR
+    }
+
+    /* Set up callback to generate checksums for updater's metadata files */
+    cb_info.func = md_ck_cb;
+
+    /* Activate private property to generate checksums for updater's metadata file */
+    H5Pset(fapl, H5F_ACS_GENERATE_MD_CK_CB_NAME, &cb_info);
+
+    /* Use file creation or file open for testing */
+    if (file_create) {
+        if ((fid = H5Fcreate(FILENAME4, H5F_ACC_TRUNC, fcpl, fapl)) < 0)
+            TEST_ERROR;
+    }
+    else {
+        fid = H5Fcreate(FILENAME4, H5F_ACC_TRUNC, fcpl, H5P_DEFAULT);
```

Comment:
Should we check the returned values of APIs in this block?

Reply:
done